src/main/java/net/glowstone/command/minecraft/GameModeCommand.java

```diff
@@ -73,22 +72,20 @@ public boolean execute(CommandSender sender, String label, String[] args) {
     private void updateGameMode(CommandSender sender, Player who, GameMode gameMode) {
         String gameModeName = GameModeUtils.prettyPrint(gameMode);
         who.setGameMode(gameMode);
+        who.setFallDistance(0);
```

Comment:
Maybe do this statement to the `GlowPlayer#setGameMode` method instead? That way, this fix would also apply to plugins changing a player's game mode.